cmd/net.go

```diff
@@ -66,18 +67,43 @@ func mustGetLocalIP4() (ipList set.StringSet) {
 // getHostIP4 returns IPv4 address of given host.
 func getHostIP4(host string) (ipList set.StringSet, err error) {
 	ipList = set.NewStringSet()
-	ips, err := net.LookupIP(host)
-	if err != nil {
-		return ipList, err
+	//variable to list resolved IP addresses
+	var ips []net.IP
+	// channel to indicate completion of host resolution
+	doneCh := make(chan struct{})
+	// prepare getElapsedTime() to calculate elapsed time since we started trying to resolve hosts.
+	formatStartTime := time.Now().Round(time.Second)
+	getElapsedTime := func() string {
+		// All times are rounded to avoid showing milli, micro and nano seconds
+		return time.Now().Round(time.Second).Sub(formatStartTime).String()
 	}
-
-	for _, ip := range ips {
-		if ip.To4() != nil {
-			ipList.Add(ip.String())
+	// Indicate to our retry routine to exit cleanly, upon this function return.
+	defer close(doneCh)
+	// Wait on the jitter retry loop
+	retryTimeCh := newRetryTimerSimple(doneCh)
+	for {
+		select {
+		case <-retryTimeCh:
+			// try to resolve remote host
+			ips, err = net.LookupIP(host)
+			if err != nil {
+				// log error and continue. We dont fail on host resolution, just log
+				// the message to console about the host not being resolveable. It is
+				// expected that user provides resolveable hosts
+				errorIf(err, "Unable to resolve host %s (elapsed %s)", host, getElapsedTime())
```

Comment:
The closure does not appear to be needed - `getElapsedTime()` could just accept the start time argument.